lab_3_generate_by_ngrams/main.py

```diff
@@ -456,7 +700,7 @@ class BackOffGenerator:
 
     Attributes:
         _language_models (dict[int, NGramLanguageModel]): Language models for next token prediction
-        _text_processor (NGramLanguageModel): A TextProcessor instance to handle text processing
+        _text_processor (TextProcessor): A TextProcessor instance to handle text processing
```

Comment:
Why did you change it?

Reply:
It seems wrong to have a text processor of a NGramLanguageModel type, it just interfers with the description, because it is by default TextProcessor type. Or isn't it? I didn't know if it's a mistake or I just don't understand the types of these inputs, so I wanted to draw attention and ask.